jaxlib/mosaic/gpu/runtime.cc

```diff
@@ -154,6 +156,20 @@ void* mosaic_gpu_module_load(void *data) {
     fprintf(stderr, "cuModuleLoadData failed: %s\n", ptr);
     abort();
   }
+
+  CUdeviceptr ptr = 0;
+  size_t size = 0;
+  // Check if module contains NVSHMEM globals implying NVSHMEM state needs to set
+  if (auto result = cuModuleGetGlobal(&ptr, &size, module,
+                                      "nvshmemi_device_lib_version_d");
+      result == CUDA_SUCCESS) {
+    if (auto result = mosaic::gpu::NvshmemApi::Default().cumodule_int(module);
+        result != NVSHMEM_SUCCESS) {
+      fprintf(stderr, "nvshmemx_cumodule_init failed.\n");
```

Comment:
Does NVSHMEM provide some analogue of `cuGetErrorString`? It would be helpful to include some more information in case this fails

Reply:
No, unfortunately.